src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/EventRoute.cs

```diff
@@ -195,25 +195,28 @@ private void InvokeHandlersImpl(object source, RoutedEventArgs args, bool reRais
 
                     if( TraceRoutedEvent.IsEnabled )
                     {
+                        _traceArguments ??= new object[3];
+                        _traceArguments[0] = _routeItemList[i].Target;
+                        _traceArguments[1] = args;
+                        _traceArguments[2] = args.Handled ? _true : _false;
                         TraceRoutedEvent.Trace(
                             TraceEventType.Start,
-                            TraceRoutedEvent.InvokeHandlers,  
-                            _routeItemList[i].Target,
-                            args,
-                            args.Handled );
-
+                            TraceRoutedEvent.InvokeHandlers,
+                            _traceArguments);
                     }
                     
                     _routeItemList[i].InvokeHandler(args);
 
                     if( TraceRoutedEvent.IsEnabled )
                     {
+                        _traceArguments ??= new object[3];
+                        _traceArguments[0] = _routeItemList[i].Target;
+                        _traceArguments[1] = args;
```

Comment:
I'm assuming that `TraceRoutedEvent.IsEnabled` can be set to true concurrently, e.g., by attaching a listener on another thread. If that's not the case, and this can't change while the method is running, these redundant lines could be deleted.

Or we could read the `IsEnabled` property once into a local.